apps/service_providers/llm_service/history_managers.py

```diff
@@ -163,8 +164,12 @@ def __init__(
         self.trace_service = session.experiment.trace_service if session else None
         self.ai_message = None
 
-        self.input_message_metadata = None
-        self.output_message_metadata = None
+        self.input_message_metadata = defaultdict(list)
+        self.output_message_metadata = defaultdict(list)
```

Comment:
not sure about using `defaultdict` here since different metadata keys might want different types